pkg/parse/root_sync_client.go

```diff
@@ -197,15 +204,32 @@ func (p *rootSyncStatusClient) SetSourceAnnotations(ctx context.Context, commit
 	err := opts.Client.Patch(ctx, rs,
 		client.RawPatch(types.MergePatchType, []byte(patch)),
 		client.FieldOwner(configsync.FieldManager))
-
 	if err != nil {
-		return fmt.Errorf("failed to patch RootSync annotations: %w\nPatch content: %s", err, patch)
+		// If denied by a webhook, assume it's for OCI signature validation, and therefore a source error.
+		if isRequestDeniedByWebhookStatusError(err) {
+			return status.SourceError.Wrap(err).Sprintf("webhook blocked patching RootSync annotation: %s", metadata.ImageToSyncAnnotationKey).Build()
+		}
+		return status.APIServerErrorf(err, "failed to patch RootSync annotation: %s", metadata.ImageToSyncAnnotationKey)
 	}
-
 	return nil
 }
 
-func (p *rootSyncStatusClient) SetRequiresRendering(ctx context.Context, renderingRequired bool) error {
+// isRequestDeniedByWebhookStatusError returns true if the error is a Kubernetes
+// *apierrors.StatusError
+func isRequestDeniedByWebhookStatusError(err error) bool {
+	if err == nil {
+		return false
+	}
+	var statusErr *apierrors.StatusError
+	if !errors.As(err, &statusErr) {
+		return false
+	}
+	// TODO: check for a specific error code to distinguish source errors, like
```

Comment:
This requires user's webhook server to implement the same contract and Config Sync would assume all webhook servers follow the contract. If the webhook server doesn't return the specific error code, it would return an API error, which is also confusing.

Reply:
If the webhook doesn't follow the interface contract, the user would be confused, yes. But that's something the user or their cluster admin can fix. 

Regardless, I've removed the TODO. If we want to use a custom status error, that can be added in a separate PR. 

This change is a compromise, not as good as a custom http code, but a solid improvement over what we have now, significantly reducing the number of possible misclassified errors.